Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/PropertyAssetCtrl.cpp

```diff
@@ -638,6 +638,21 @@ namespace AzToolsFramework
         {
             UpdateAssetDisplay();
         }
+        else if (!m_selectedAssetID.IsValid())
+        {
+            // If we get here without a valid assetId, it means the user has pressed the create button and we're waiting for an asset to be created.
+            // Use it if it's of the correct type.
+            AzFramework::AssetCatalogEventBus::Handler::BusDisconnect();
+
+            AZ::Data::AssetInfo assetInfo;
+            AZ::Data::AssetCatalogRequestBus::BroadcastResult(
+                assetInfo, &AZ::Data::AssetCatalogRequestBus::Events::GetAssetInfoById, assetId);
+
+            if (assetInfo.m_assetType == GetCurrentAssetType())
+            {
+                SetSelectedAssetID(assetId);
+            }
+        }
```

Comment:
This logic is fuzzy.
There is no guarantee that the Asset being added is the one the that was actually created.
The relative filename of the assetInfo should also be checked to validate that the asset matches.

It is very easy for someone to drag-and-drop an asset of the monitored AssetType to a Scan Folder when the Editor is running and this function will be invoked with.

Let's say there is a A Component with an field that accepted an InputEventBindingsAsset that used this property handler.

If a random `*.inputbindings` asset was added to any monitored folder by the AP, then this property handler would update to use that new asset.

Reply:
For example if the source such as the [thirdpersonmovement.inputbindings](https://github.com/o3de/o3de/tree/development/Gems/StartingPointInput/Assets/thirdpersonmovement.inputbindings) was copied into let's the AutomatedTesting/Assets directory, then this will cause all PropertyAssetCtrl's which handle `InputEventBindingsAsset` type will have it's selected asset ID re-assigned to it.

Reply:
@sphrose Ping related to requested changes on PR

Reply:
Sorry for the delay, you're quite right, I hadn't considered the dropping assets issue. I'll look into it.

Reply:
@lumberyard-employee-dm Could you check your issues have been addressed in the latest version please?

Reply:
I am checking now.

Reply:
Well this particular issue has been addressed, as the new OnAssetCreated event will send the Asset ID for the specific created new asset to the PropertyAssetCtrl widget, but there is another issue that has now surfaced with the changes which is detailed in this comment: https://github.com/o3de/o3de/pull/6508#discussion_r1061135896